src/contexts/bridge.context.tsx

```diff
@@ -0,0 +1,118 @@
+import { BigNumber, ContractTransaction, PayableOverrides } from "ethers";
+import { createContext, FC, useContext, useEffect, useState } from "react";
+
+import { useEnvContext } from "src/contexts/env.context";
+import { useProvidersContext } from "src/contexts/providers.context";
+import { Bridge, Bridge__factory } from "src/types/contracts/bridge";
+
+interface BridgeContext {
+  bridge: (
+    token: string,
+    amount: BigNumber,
+    destinationNetwork: number,
+    destinationAddress: string,
+    overrides?: PayableOverrides & { from?: string | Promise<string> }
+  ) => Promise<ContractTransaction>;
+  claim: (
+    originalTokenAddress: string,
+    amount: BigNumber,
+    originalNetwork: string,
+    destinationNetwork: number,
+    destinationAddress: string,
+    smtProof: string[],
+    globalExitRootNum: number,
+    mainnetExitRoot: string,
+    rollupExitRoot: string
+  ) => Promise<ContractTransaction>;
+}
+
+const bridgeContextNotReadyErrorMsg = "The bridge context is not yet ready";
+
+const bridgeContext = createContext<BridgeContext>({
+  bridge: () => {
+    return Promise.reject(bridgeContextNotReadyErrorMsg);
+  },
+  claim: () => {
+    return Promise.reject(bridgeContextNotReadyErrorMsg);
+  },
+});
+
+const BridgeProvider: FC = (props) => {
+  const env = useEnvContext();
+  const { connectedProvider } = useProvidersContext();
+  const [bridgeContract, setBridgeContract] = useState<Bridge>();
+
+  const bridge = (
+    token: string,
+    amount: BigNumber,
+    destinationNetwork: number,
+    destinationAddress: string,
+    overrides?: PayableOverrides & { from?: string | Promise<string> }
+  ): Promise<ContractTransaction> => {
+    if (env === undefined) {
+      throw new Error("Environment is not available");
+    }
+
+    if (bridgeContract === undefined) {
+      throw new Error("Bridge contract is not available");
+    }
+
+    return bridgeContract.bridge(token, amount, destinationNetwork, destinationAddress, overrides);
+  };
+
+  const claim = async (
```

Comment:
Shouldn't this and `bridge` functions use the `useCallback` hook as they rely on state from other hooks?

Reply:
I just enter the PR to comment on this 😄